expo/eas-cli
# [eas-cli] [eas-json] [ENG-10069] add support for bun

# Why

Bun is fast, and people want to use it! https://bun.sh/

# Related PRs:

1. `eas-build`: https://github.com/expo/eas-build/pull/279
2. `turtle-v2`: https://github.com/expo/turtle-v2/pull/1424 (deploy FIRST)
3. 👉 `eas-cli` https://github.com/expo/eas-cli/pull/2055 (deploy LAST - depends on turtle and universe)
4.  `universe` https://github.com/expo/universe/pull/13473 (deploy SECOND)

# Testing
1. checkout `@kadikraman/bun-support` in `universe` & in `universe/server/www` run `yarn start:docker`
2. on the same branch in `universe/server/website`, run `yarn start:local`
3. open http://expo.test/ and log in / create account if you haven't yet
4. checkout `@kadikraman/support-bun-version-in-eas-json` in `turtle-v2`, run `yarn` and run `yarn docker:up`
5. on the same branch, at the root level of `turtle-v2` run `yarn start`
6. comment out this line https://github.com/expo/turtle-v2/blob/main/src/services/worker/src/runtimeEnvironment.ts#L57 (else the custom tools step is skipped for local builds)
7. checkout `@kadikraman/bun-support` in `eas-cli`, run `yarn` and `yarn build`
Open your .zshrc or other config file and add:
`alias easd="/PATH/TO/eas-cli/bin/run"` (replace `PATH/TO` with your path)
8. in your favourite demo app and add `"bun": "1.0.0"`
9. if you have a `yarn.lock` or `package-lock.json`, delete it and run `bun install` to create a `bun.lockb` (if you have a `yarn.lock`, then `yarn` is used by default) - [context](https://github.com/expo/expo/blob/142d0b5531ef7b76208e388f74ddf43648510ad5/packages/%40expo/package-manager/src/utils/nodeManagers.ts#L28)
10. in the demo app, run `EXPO_LOCAL=1 easd build --profile development:simulator` (or your equivalent profile)
11. 🥟 verify that under "install dependencies" they are installed with bun@1.0.1
